packages/@aws-cdk/core/lib/assets.ts

```diff
@@ -60,13 +60,10 @@ export interface DockerImageAssetSource {
   /**
    * ECR repository name
    *
-   * Specify this property if you need to statically address the image, e.g.
-   * from a Kubernetes Pod. Note, this is only the repository name, without the
-   * registry and the tag parts.
-   *
-   * @default - automatically derived from the asset's ID.
+   * Note, this is only the repository name, without the registry and
+   * the tag parts.
    */
-  readonly repositoryName?: string;
+  readonly repositoryName: string;
```

Comment:
You are right we can keep this optional but as it is now `repositoryName` will always be defined when calling `addDockerImageAsset`, no? You want to avoid the breaking change?

Reply:
Yes, I want to avoid the breaking change, and also, `addDockerImageAsset` is lower-level and we might introduce another `ImageAsset` as we implement the new CI/CD support.

Reply:
If make this back optional you either have to make it also optional in `ContainerImageAssetMetadataEntry` and then `throw` if undefined in `prepareEcrRepository`? or set a default when defining `metadata`, what default and how would you document it because you don't have access to the `asset` node there (I would leave the default behavior on `repositoryName` in `DockerImageAsset`)?

Reply:
I believe it was already optional in `ContainerImageAssetMetadataEntry`, no?